File: codetest-python/step3.py
alpha = ['B','W','O','G','Y','R']
arr = []

# 큐브 3차원 리스트 생성
for i in range(len(alpha)):
    n = [ [0] * 3 for m in range(3)]
    for j in range(3):
        for k in range(3):
            n[j][k] = alpha[i]
    arr.append(n)

# 큐브 상태 출력
for i, value in enumerate(arr):
    for j, value1 in enumerate(range(3)):
        for k, value2 in enumerate(range(3)):
            print(arr[i][j][k], end=" ")
        print(' ')
    print('')

File: codetest-java/src/step3/Step3.java
package step3;

public class Step3 {
    char[] alpha = {'B', 'W', 'O', 'G', 'Y', 'R'};
    char arr[][][] = new char[6][3][3];

    // 큐브 상태 출력
    void printcube(){
        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < alpha.length; j++) {
                for (int k = 0; k < 3; k++) {
                    System.out.print(alpha[j]+"\t");
                }
                if (j==5) {
                    System.out.println("");
                }else{
                    System.out.print("\t\t");
                }
            }
        }
    }
    //  3차원 배열에 큐브상태 저장
    void savecube(){
        for(int i=0; i<alpha.length; i++){
            for(int j=0; j<3; j++){
                for(int k=0; k<3; k++){
                    arr[i][j][k] = alpha[i];
                }
            }
        }
    }
    public static void main(String[] args) {
        Step3 s = new Step3();
        s.savecube();
        s.printcube();
    }
}
